src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,208 @@
+                    + car + ". ", e);
+        }
+    }
+
+    @Override
+    public Car get(Long id) {
+        String query = "SELECT id, model, manufacturer_id, driver_id "
+                + "FROM cars c JOIN cars_drivers cd "
+                + "ON c.id = cd.car_id where is_deleted = false AND c.id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            Car car = null;
+            if (resultSet.next()) {
+                car = getCar(resultSet);
+            }
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get car by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT * FROM cars WHERE is_deleted = FALSE";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get a list of cars from carsDB.", e);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        Long carId = car.getId();
+        String query = "UPDATE cars "
+                + "SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                        = connection.prepareStatement(query)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.setLong(3, carId);
+            statement.executeUpdate();
+
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update "
+                    + car + " in carsDB.", e);
+        }
+        List<Driver> driversForCar = car.getDrivers();
+        List<Driver> currentDrivers = getDriversForCar(carId);
+        if (driversForCar.size() > currentDrivers.size()) {
+            for (Driver driver : driversForCar) {
+                if (!currentDrivers.contains(driver)) {
+                    insertDriverToCar(carId, driver.getId());
+                }
+            }
+        } else if (driversForCar.size() < currentDrivers.size()) {
+            for (Driver driver : currentDrivers) {
+                if (!driversForCar.contains(driver)) {
+                    deleteDriverFromCar(carId, driver.getId());
+                }
+            }
+        }
```

Comment:
It would be better to put this logic into separate methods and call them, for example: isertDriversToCar(), deleteDriversFromCar()